stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2435,25 +2435,31 @@ function _sqrt_quasitriu_offdiag_block!(R, A)
     return R
 end
 
+# real square root of 2x2 diagonal block of quasi-triangular matrix from real Schur
+# decomposition. Eqs 6.8-6.9 and Algorithm 6.5 of
+# Higham, 2008, "Functions of Matrices: Theory and Computation", SIAM.
 Base.@propagate_inbounds function _sqrt_real_2x2!(R, A)
     a11, a21, a12, a22 = A[1, 1], A[2, 1], A[1, 2], A[2, 2]
+    d = a11 - a22
     θ = (a11 + a22) / 2
-    μ² = -(a11 - a22)^2 / 4 - a21 * a12
-    μ = sqrt(μ²)
-    if θ > 0
-        α = sqrt((sqrt(θ^2 + μ²) + θ) / 2)
-    else
-        α = μ / sqrt(2 * (sqrt(θ^2 + μ²) - θ))
-    end
+    μ = sqrt(-d^2 - 4 * a21 * a12) / 2
+    α = _real_sqrt(θ, μ)
     c = 2α
-    d = α - θ / c
-    R[1, 1] = a11 / c + d
+    f = d / 4α
+    R[1, 1] = α + f
     R[2, 1] = a21 / c
     R[1, 2] = a12 / c
-    R[2, 2] = a22 / c + d
+    R[2, 2] = α - f
     return R
 end
 
+# real part of square root of θ+im*μ
+@inline function _real_sqrt(θ, μ)
+    iszero(θ) && iszero(μ) && return sqrt(zero(θ)^2 + zero(μ)^2)
```

Comment:
Why is this special case needed?   Won't you get `t=0` and return `t` (from the `θ ≥ 0` branch) below?

Reply:
Yes, this is just for efficiency, because we can then skip a `sqrt` and `hypot` call. Perhaps something like this is more useful.

Reply:
Is it common enough to be worth the cost of the extra checks in the nonzero cases?

Reply:
Actually, come to think of it, we should never hit this branch. If the 2x2 block has eigenvalues equal to 0, then it's the zero matrix, and it's not a 2x2 block.